specs/asgi.rst

```diff
@@ -179,14 +184,24 @@ implementations, as a class::
         async def __call__(self, receive, send):
             ...
 
+        async def startup(self):
```

Comment:
You have written these as methods on `self`, but they would in fact need to be `classmethod`s, no? The application object isn't instantiated at server start.

Reply:
Yep, this is a silly mistake.